File: gymgenie-mobile/iosApp/iosApp/Components/WorkoutTabSelector.swift
import SwiftUI
import Shared

struct WorkoutTabSelector: View {
    let selectedTab: Shared.WorkoutsTab
    let onTabSelected: (Shared.WorkoutsTab) -> Void

    private let deepInk = Color(red: 0.161, green: 0.141, blue: 0.125)
    private let mutedText = Color(red: 0.463, green: 0.447, blue: 0.416)
    private let pillBg = Color(red: 0.929, green: 0.925, blue: 0.918) // #EDECEA

    var body: some View {
        HStack(spacing: 3) {
            tabButton(title: "Мои планы", tab: .workouts)
            tabButton(title: "Каталог", tab: .exercises)
        }
        .padding(3)
        .background(Capsule().fill(pillBg))
        .padding(.horizontal, 20)
        .padding(.vertical, 8)
    }

    private func tabButton(title: String, tab: Shared.WorkoutsTab) -> some View {
        let selected = selectedTab == tab
        return Button(action: {
            withAnimation(.easeInOut(duration: 0.2)) { onTabSelected(tab) }
        }) {
            Text(title)
                .font(.system(size: 14, weight: .semibold))
                .foregroundColor(selected ? .white : mutedText)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 10)
                .background(Capsule().fill(selected ? deepInk : Color.clear))
        }
        .buttonStyle(.plain)
    }
}

File: gymgenie-mobile/composeApp/src/androidMain/kotlin/com/asc/gymgenie/feature/workouts/components/WorkoutTabSelector.kt
package com.asc.gymgenie.feature.workouts.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.asc.gymgenie.presentation.WorkoutsTab
import com.asc.gymgenie.ui.theme.DeepInk
import com.asc.gymgenie.ui.theme.MutedText
import com.asc.gymgenie.ui.theme.PillBg

@Composable
fun WorkoutTabSelector(
    selectedTab: WorkoutsTab,
    onTabSelected: (WorkoutsTab) -> Unit,
    modifier: Modifier = Modifier,
) {
    val tabs = listOf(
        WorkoutsTab.WORKOUTS to "Мои планы",
        WorkoutsTab.EXERCISES to "Каталог",
    )
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(50))
            .background(PillBg)
            .padding(3.dp),
        horizontalArrangement = Arrangement.spacedBy(0.dp),
    ) {
        tabs.forEach { (tab, label) ->
            val selected = selectedTab == tab
            Box(
                modifier = Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(50))
                    .background(if (selected) DeepInk else Color.Transparent)
                    .clickable { onTabSelected(tab) }
                    .padding(vertical = 10.dp),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    text = label,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = if (selected) Color.White else MutedText,
                )
            }
        }
    }
}
